### PR TITLE
PIMS-1404 Agency Details

### PR DESCRIPTION
**Pull request by dbarkowsky**
## 🎯 Summary

  
[PIMS-1404](https://citz-imb.atlassian.net/browse/PIMS-1404?atlOrigin=eyJpIjoiMzAxZmNmMzFlN2JiNGY0NmI1MDY5YmMxMWQ5ZTVlNjgiLCJwIjoiaiJ9)

## Changes
- Add Agencies Details page, which includes:
  - Editing agency details and notification settings
  - Deleting agency (only if there are no FK constraints. We need a more user-friendly outcome for these)
- useAgencies API hooks

## Testing
- Navigate to the Administration > Agencies page. `/admin/agencies`
- Select an agency and try editing, etc.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: react-app/src/components/agencies/AgencyDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  };
+
+  const notificationsSettingsData = {
+    SendEmail: data?.SendEmail,
+    To: data?.Email ? data.Email.split(';') : [],
+    CC: data?.CCEmail ? data.CCEmail.split(';') : [],
+  };
+
+  const customFormatter = (key: keyof AgencyStatus, val: any) => {
+    console.log(val);
```
Comment (GrahamS-Quartech) [inline react-app/src/components/agencies/AgencyDetails.tsx:58]: Remove

**Comment by GrahamS-Quartech** [comment]
Couple things I noticed:
* The email input works nicely with the enter key but I was expecting it to do something similar when I pressed comma or semi-colon... Instead, it will just eat the input but do nothing with it when you hit submit. I think forcing the use of enter key could be fine if it's indicated somehow, but I think better would be allowing enter, comma, or semi-colon.
* You can set the agency's parent to itself, which I imagine isn't valid? But doesn't seem to cause an error necessarily.

**Comment by dbarkowsky** [comment]
> Couple things I noticed:
> 
> * The email input works nicely with the enter key but I was expecting it to do something similar when I pressed comma or semi-colon... Instead, it will just eat the input but do nothing with it when you hit submit. I think forcing the use of enter key could be fine if it's indicated somehow, but I think better would be allowing enter, comma, or semi-colon.
> * You can set the agency's parent to itself, which I imagine isn't valid? But doesn't seem to cause an error necessarily.

- Email chips should now delimit via comma or semicolon as well. The secret was not specifying a key for the TextField and then also having the value attached to inputValue on the Autocomplete, not the TextField.
- An agency's parent can no longer be set to itself or one of its current children. Marked those options as disabled in the autocomplete.
- Still need to validate email input, but I'm thinking that is part of the future ticket for zod validation.

**Comment by GrahamS-Quartech** [comment]
>     * Still need to validate email input, but I'm thinking that is part of the future ticket for zod validation.

Yeah this was the only thing I was gonna comment on, otherwise all works nicely now. Approving.

**Review by GrahamS-Quartech — APPROVED** [review]

**Pull request merged**